### PR TITLE
feat(integration/dagster):  dagster connector to emit dagster pipeline

### PR DESCRIPTION
**Pull request by shubhamjagtap639**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by treff7es** [comment]
@shubhamjagtap639, please, can you fix the linter errors?

**Comment by shubhamjagtap639** [comment]
> @shubhamjagtap639, please, can you fix the linter errors?

@treff7es There is no linter errors. CI is getting failed bacause of below error:

ERROR: Cannot install apache-airflow[snowflake]==2.0.2, apache-airflow[snowflake]==2.1.0, apache-airflow[snowflake]==2.1.1, apache-airflow[snowflake]==2.1.2, apache-airflow[snowflake]==2.1.3, apache-airflow[snowflake]==2.1.4, apache-airflow[snowflake]==2.2.5, apache-airflow[snowflake]==2.3.0, apache-airflow[snowflake]==2.3.1, apache-airflow[snowflake]==2.3.2, apache-airflow[snowflake]==2.3.3, apache-airflow[snowflake]==2.3.4, apache-airflow[snowflake]==2.4.0, apache-airflow[snowflake]==2.4.1, apache-airflow[snowflake]==2.4.2, apache-airflow[snowflake]==2.4.3, apache-airflow[snowflake]==2.5.0, apache-airflow[snowflake]==2.5.1, apache-airflow[snowflake]==2.5.2, apache-airflow[snowflake]==2.5.3, apache-airflow[snowflake]==2.6.0, apache-airflow[snowflake]==2.6.1, apache-airflow[snowflake]==2.6.2, apache-airflow[snowflake]==2.6.3 and apache-airflow~=2.2.0 because these package versions have conflicting dependencies.

**Comment by mohdsiddique** [comment]
Hi @treff7es 
@shubhamjagtap639 did the investigation on conflicting dependency issues and below are the findings. 

The dagster connector depends on:
-  dagster>=1.3.3 and 
- dagster depends on typing-extensions>=4.4.0

The conflict dependency issue arises from snowflake connector, where snowflake connector depends on:
- acryl-datahub-classify==0.0.8 and
- acryl-datahub-classify depends on spacy>=3.4.1,<=3.4.3 and 
- spacy depends on typing_extensions>=3.7.4.1,<4.2.0; python_version < "3.8".

so here the issue is the package `typing-extensions`, dagster needs it to be `typing-extensions>=4.4.0` but spacy restricting it to be `typing-extensions<4.2.0` for python version < "3.8" and hence our CI is passing for Python3.10 and *failing for 3.7*.

There are possible two solutions: Either one of we need to apply 
1. Set the max range of spacy in `acryl-datahub-classify` to `<=3.5.0` which is currently `<=3.4.3`
2. Lower the version of dagster 

On #1: @mayurinehate or @Krushnakant Mardikar can comments. We did install the spacy 3.5.0 in our local environment and `./gradlew :metadata-ingestion:installDev` worked fine. 

On #2: The dagster lower version `1.1.20` depends on `typing-extensions>=4.0.0` which resolves this conflicting dependency issue but the dagster version `1.1.20` is missing critical functions listed below which we are using to extract dagster job and ope metadata. 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Missing functions in dagster `1.1.20`: 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1. get_job_snapshots
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2. get_run_stats
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3. get_execution_plan_snapshot     
We do need to put more effort into #2 as we are moving to an older package version and need to find out alternate methods to perform the same task

**Comment by mardikark-gslab** [comment]
Tested the datahub-classify functionality by changing Spacy version upper limit to 3.5.0, all tests passed. Created a pull request with this change.
https://github.com/acryldata/datahub-classify/pull/19

**Comment by mohdsiddique** [comment]
> Tested the datahub-classify functionality by changing Spacy version upper limit to 3.5.0, all tests passed. Created a pull request with this change. [acryldata/datahub-classify#19](https://github.com/acryldata/datahub-classify/pull/19)
> 
> cc: @mayurinehate @hsheth2

@mardikark-gslab 
I can see some failures at [CI](https://github.com/acryldata/datahub-classify/actions/runs/5595861605/jobs/10232187573?pr=19). Are they acceptable?

File: metadata-ingestion/setup.py [hunk 1]
```diff
@@ -392,6 +392,9 @@ def get_long_description():
     "powerbi-report-server": powerbi_report_server,
     "vertica": sql_common | {"vertica-sqlalchemy-dialect[vertica-python]==0.0.8"},
     "unity-catalog": databricks | sqllineage_lib,
+    "dagster": {
```
[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/setup.py:395]: cc: @hsheth2

[the remark this answers is not in the document]
Comment (hsheth2) [reply metadata-ingestion/setup.py]: yup agreed

**Pull request closed without merging**